src/flow/src/expr/func.rs

```diff
@@ -221,6 +221,57 @@ impl UnaryFunc {
         }
     }
 
+    pub fn from_tumble_func(name: &str, args: &[TypedExpr]) -> Result<(Self, TypedExpr), Error> {
+        match name {
+            "tumble_start" | "tumble_end" => {
+                let ts = args.first().context(InvalidQuerySnafu {
+                    reason: "Tumble window function requires a timestamp argument",
+                })?;
+                let window_size = args
+                .get(1)
+                .and_then(|expr| expr.expr.as_literal())
+                .context(InvalidQuerySnafu {
+                    reason: "Tumble window function requires a window size argument"
+                })?.as_string() // TODO(discord9): since df to substrait convertor does not support interval type yet, we need to take a string and cast it to interval instead
+                .map(|s|cast(Value::from(s), &ConcreteDataType::interval_month_day_nano_datatype())).transpose().map_err(BoxedError::new).context(
+                    ExternalSnafu
+                )?.and_then(|v|v.as_interval())
+                .with_context(||InvalidQuerySnafu {
+                    reason: format!("Tumble window function requires window size argument to be a string describe a interval, found {:?}", args.get(1))
+                })?;
```

Comment:
Don't know why rustfmt won't handle this paragraph...

Reply:
> Don't know why rustfmt won't handle this paragraph...

This: https://github.com/rust-lang/rustfmt/issues/3863#issuecomment-700441739
found many other fmt errors too.... `cargo fmt -- --config error_on_line_overflow=true;`